src/main/java/edu/cornell/kfs/fp/batch/service/impl/CreateAccountingDocumentServiceImpl.java

```diff
@@ -236,14 +323,28 @@ protected void removeDoneFileQuietly(String dataFileName) {
     
     protected void createAndEmailReport(CreateAccountingDocumentReportItem reportItem) {
         createAccountingDocumentReportService.generateReport(reportItem);
-        String toAddress;
+        String primaryToAddress;
         String fromAddress = getCreateAccountingDocumentReportEmailAddress();
         if (reportItem.isXmlSuccessfullyLoaded()) {
-            toAddress = reportItem.getReportEmailAddress();
+            primaryToAddress = reportItem.getReportEmailAddress();
+        } else {
+            primaryToAddress = fromAddress;
+        }
+        
+        List<String> toAddresses;
+        if (!reportItem.isXmlSuccessfullyLoaded() && reportItem.isDuplicateFile()
+                && shouldExcludePreviouslyProcessedFiles()) {

```

Comment:
Should this boolean expression just be reportItem.isDuplicateFile()?  We only send email in production, and we will always prevent duplicate in prod.  Also, if duplicates are are found, XML won't fully be loaded, correct?

Reply:
I think we need at least the `isDuplicateFile()` and `shouldExcludePreviouslyProcessedFiles()` checks here. If the parameter is configured to allow duplicate processing, the report item will be marked as a duplicate but KFS will process it anyway. I don't think we want the extra duplicate-file recipients to be notified if KFS has been configured to allow such duplication. If you think otherwise, please let me know.

I believe I could at least simplify this to remove the `!reportItem.isXmlSuccessfullyLoaded()` check.